### ISSUE TITLE
Persistent builds don't seem to work

### ISSUE THREAD

**Issue by lzybkr** [nightroman/Invoke-Build#140]
It looks like `XTask` is invoked before the task is [run](https://github.com/nightroman/Invoke-Build/blob/b97ea9111147c4ce35d29165a4241793f6959ea3/Invoke-Build.ps1#L478), so `Elapsed` is never set when read [here](https://github.com/nightroman/Invoke-Build/blob/b97ea9111147c4ce35d29165a4241793f6959ea3/Build-Checkpoint.ps1#L77).

Repro:

```powershell
﻿
param($stuff)

task A {
    Write-Host "task A"
}

task B A,{
    $script:stuff = "Does this persist?"
    Write-Host "task B"
    if (!(Test-Path aa.txt)) {
         exec { cmd /c exit 1 }
    }
}

task C B,{
    Write-Host "task C: $stuff"
}

task . C
```

```powershell
Build-Checkpoint tmp.clixml -Build @{File = '.\a.build.ps1' }
"" > aa.txt
Build-Checkpoint tmp.clixml -resume
```

Expected - it shouldn't run task A the second time.

**Comment by nightroman** [comment]
Hi @lzybkr 

Weird... I thought it was fine to invoke `XTask` before tasks (and there was a reason to invoke before). In this case it should be invoked before B after A, so that and A's `Elapsed` should be set and A should be saved as done. I'll take a closer look indeed.

**Comment by nightroman** [comment]
> In this case it should be invoked before B after A...

Ah, it's wrong. Due to the dependencies, `XTask` for B is invoked before A. It's a bug. Thank you for catching and reporting it.

**Comment by nightroman** [comment]
Here is the story of invoking `XTask` before tasks -- https://github.com/nightroman/Invoke-Build/issues/22

**Comment by nightroman** [comment]
Fixed in 5.4.3
